sunshine/src/main/java/io/github/tatools/sunshine/FileConfig.java

```diff
@@ -9,20 +11,41 @@
  */
 public class FileConfig implements Config {
 
-    private final InputStream file;
+    private final Wrapper wrapper;
 
     public FileConfig(InputStream file) {
-        this.file = file;
+        this.wrapper = new Wrapper(file);
     }
 
     @Override
     public String property(String key) {
-        try {
-            Properties property = new Properties();
-            property.load(file);
-            return property.getProperty(key, "not defined");
-        } catch (java.io.IOException e) {
-            throw new IOException(e);
+        return wrapper.get().getProperty(key);
+    }
+
+    @Override
+    public boolean has(String key) {
+        return wrapper.get().getProperty(key) != null;
+    }
+
+    private class Wrapper {
+        final List<Properties> props = new ArrayList<>();
+        final InputStream from;
+
+        private Wrapper(InputStream from) {
+            this.from = from;
+        }
+
+        Properties get() {
+            if (props.isEmpty()) {
+                try {
+                    Properties property = new Properties();
+                    property.load(from);
+                    props.add(property);
+                } catch (java.io.IOException e) {
+                    throw new IOException(e);
+                }
+            }
+            return props.get(0);
```

Comment:
This your approach I will have mutable object and null, which both are bad practices.